app/components/security.js

```diff
@@ -32,8 +32,24 @@ class Security {
                 securityCookie = req.cookies[SECURITY_COOKIE];
             }
 
-            // Retrieve user details
             if (securityCookie) {
+                const lostSession = !req.session.expires;
+                const sessionExpired = req.session.expires?req.session.expires <= Date.now():false;
+                if (lostSession || sessionExpired) {
+                    if (lostSession) {
+                        logger.error('The current user session is lost.');
+                    } else {
+                        logger.error('The current user session has expired.');
+                    }
+                    logger.info('Redirecting user to the time-out page.');
+                    res.clearCookie(SECURITY_COOKIE);
+                    delete req.cookies[SECURITY_COOKIE];
+                    return res.redirect('/time-out');
+                }
+
+                logger.debug('Extending session for active user.');
+                req.session.expires = Date.now() + config.app.session.expires;
+
```

Comment:
We already have the logger on the request and this adds the session id which will help track logging for a specific user, for example use 

`req.log.error('The current user session is lost.');` 

instead of 

`logger.error('The current user session is lost.');`

This will display a log message like this

`2018-11-06T13:55:17+00:00 - error: [frontend, sessionId: ZGy_k7uHQ_9TmMHTdQ6vbGXx69-TtfnQ] Document upload timed out`

instead of 

`2018-11-06T13:55:17+00:00 - error: [frontend, sessionId: Init] Document upload timed out`